front/pages/w/[wId]/members/index.tsx

```diff
@@ -560,35 +560,35 @@ function InviteEmailModal({
         <div>Do you want to proceed?</div>
       </div>
     );
+    const existingMembersArePresent =
+      activeDifferentRole.length > 0 || revoked.length > 0;
 
-    if (!shouldWarnAboutExistingMembers(invitesByCase)) {
-      await sendInvitations({
-        owner,
-        emails: notInWorkspace,
-        invitationRole,
-        sendNotification,
-      });
-      await mutate(`/api/w/${owner.sId}/invitations`);
-      onClose();
-    } else {
-      const shouldProceed = await confirm({
+    const shouldProceedWithInvites =
+      !existingMembersArePresent ||
+      (await confirm({
         title: "Some users are already in the workspace",
         message: ReinviteUsersMessage,
         validateLabel: "Yes, proceed",
         validateVariant: "primaryWarning",
-      });
+      }));
 
-      if (shouldProceed) {
+    if (shouldProceedWithInvites) {
+      if (existingMembersArePresent) {
         await handleMembersRoleChange({
           members: [...activeDifferentRole, ...revoked],
           role: invitationRole,
-          mutate,
           sendNotification,
         });
-
-        await mutate(`/api/w/${owner.sId}/invitations`);
-        onClose();
       }
+      await sendInvitations({
+        owner,
+        emails: notInWorkspace,
+        invitationRole,
+        sendNotification,
+      });
+      await mutate(`/api/w/${owner.sId}/members`);
```

Comment:
not sure we need 2 mutates?

Reply:
You mean on `members` and one on `invitations`?

Reply:
Let's at least gate the one on members behind `hasExistingMembers`